src/elements/text-element.js

```diff
@@ -59,4 +59,30 @@ export default class TextElement extends TextualElement {
   get hasPattern() {
     return this.pattern && this.pattern.length;
   }
+
+  get hasPatternDescription() {
+    return this.patternDescription && this.patternDescription.length;
+  }
+
+  toJSON() {
+    const json = super.toJSON();
+
+    json.numeric = this._isNumeric;
```

Comment:
`this.isNumeric` for consistency with the line below?

Reply:
Done